lib/src/component_declaration/builder_helpers.dart

```diff
@@ -120,9 +124,61 @@ abstract class UiProps extends component_base.UiProps with GeneratedClass {
       ' Will be removed in 4.0.0.')
   @toBeGenerated String get propKeyNamespace => throw UngeneratedError(member: #propKeyNamespace);
 
+  @toBeGenerated PropsMetaCollection get $meta => throw UngeneratedError(member: #meta);
+
   @override @toBeGenerated Map get props => throw UngeneratedError(member: #props);
 }
 
+extension UiPropsMeta on UiProps {
+  static const consumedPropsAssertMessage = 'consumedProps must be either an Iterable<PropsMeta> or PropsMeta';
+
+  /// A collection of metadata for the prop fields in all prop mixins used by
+  /// this props instance's generated props class.
+  ///
+  /// Synonymous with [UiComponent2]'s `propsMeta`.
+  ///
+  /// This can be used to derive consumed props by usage in conjunction with [addUnconsumedProps]
+  /// and [addUnconsumedDomProps].
+  PropsMetaCollection get staticMeta => $meta;
+
+
+  /// Copies props from the provided [props] instance to this [UiProps] instance,
+  /// filtering out props found in [consumedProps].
+  ///
+  /// [consumedProps] should be a `Iterable<PropsMeta>` instance.
+  /// This is the return type of [PropsMetaCollection]'s related APIs `forMixins`
+  /// and `allExceptForMixins`. `forMixin` can be used easily by then calling `.inList()`
+  /// on the returned [PropsMeta] instance.
```

Comment:
Like I commented above, this shouldn't mention `forMixin` and should maybe mention `.all`.

Also, I think it'd be helpful to mention this is useful for forwarding props in function components, and to provide a small example showing how it will be used in the real world.

I think the example can be fairly simple, and doesn't need all the boilerplate and stuff; something like:
```dart
// Consider props in FooProps "consumed"...
final consumedProps = props.staticMeta.forMixins({FooProps});
// ...and filter them out when forwarding props to Bar.
return (Bar()..addUnconsumedProps(props, consumedProps))();
```